src/main/java/Duke.java

```diff
@@ -2,6 +2,7 @@
 
 public class Duke {
```

Comment:
There are a lot of implementation details, perhaps it would be even better if you could refactor and abstract most of it into other methods so that other readers have an easier time reading it.